python/ray/data/tests/test_preprocessors.py

```diff
@@ -1631,18 +1608,16 @@ def test_arrow_pandas_support_pandas_dataset(create_dummy_preprocessors):
 
     assert with_pandas.transform(ds)._dataset_format() == "pandas"
 
-    assert with_arrow.transform(ds)._dataset_format() == "arrow"
-
-    assert with_pandas_and_arrow.transform(ds)._dataset_format() == "pandas"
+    assert with_pandas_and_numpy.transform(ds)._dataset_format() == "pandas"
 
 
-def test_arrow_pandas_support_arrow_dataset(create_dummy_preprocessors):
+def test_numpy_pandas_support_arrow_dataset(create_dummy_preprocessors):
```

Comment:
We should also have coverage for a "tensor dataset", i.e. a dataset created from `ray.data.from_numpy()` that's represented under the hood with a single-tensor-column table, and that should go through the NumPy narrow waist automatically. This would only need to work with the `with_numpy` and `with_pandas_and_numpy` preprocessors, I believe.